WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -57,7 +58,35 @@ public async Task<InputRegistrationResponse> RegisterInputAsync(InputRegistratio
 			// not say anything about GUID version or randomness source,
 			// only that the probability of duplicates is very low).
 			var id = new Guid(Random.GetBytes(16));
-			var alice = new Alice(coin, request.OwnershipProof, round, id);
+
+			var isPayingZeroCoordinationFee = InMemoryCoinJoinIdStore.Contains(coin.Outpoint.Hash);
+
+			if (!isPayingZeroCoordinationFee)
+			{
+				// If the coin comes from a tx that all of the tx inputs are coming from a CJ (1 hop - no pay).
+				Transaction? tx = null;
+				try
+				{
+					tx = await Rpc.GetRawTransactionAsync(coin.Outpoint.Hash, true, cancellationToken).ConfigureAwait(false);
+
+					// Cannot return null, because throwIfNotFound set to true.
+					if (tx.Inputs.All(input => InMemoryCoinJoinIdStore.Contains(input.PrevOut.Hash)))
+					{
+						isPayingZeroCoordinationFee = true;
+					}
+				}
+				catch (Exception ex)
```

Comment:
I added this because this is an extra check after `OutpointToCoinAsync` - so if that passed, here we just try to determine if it will pay or not. However, as you mentioned it is clearly not a good sign if this fails.

Reply:
Removed the try/catch